onnxruntime/core/providers/iree/compiler/torch-mlir-import-onnx/OnnxImporter.cpp

```diff
@@ -973,8 +1012,14 @@ Status NodeImporter::ImportConstantOfShapeNode(const onnx::NodeProto &node) {
 
 Status NodeImporter::GetImmediateShapeTensor(const std::string &name,
                                              std::vector<int64_t> &shape) {
-  const onnx::TensorProto &tp =
-      *graph_info_.graph_viewer().GetConstantInitializer(name, false);
+  const onnx::TensorProto *tensor =
+      graph_info_.graph_viewer().GetConstantInitializer(name, false);
+  if (!tensor) {
+    return SetError(
+        "Could not find immediate shape tensor in graph initializers");
```

Comment:
nit: add an or any before "immediate shape tensor"?

Reply:
Added a commit to emit the name of the tensor in the diagnostic, with a better error message.